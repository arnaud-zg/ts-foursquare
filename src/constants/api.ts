export const CLIENT_ID_KEY = 'apiClientId'
export const CLIENT_SECRET_KEY = 'apiClientSecret'

export interface ICredentials {
  [CLIENT_ID_KEY]: string
  [CLIENT_SECRET_KEY]: string
}

export enum EApiDefaultParameters {
  VERSION = '20190425',
}
