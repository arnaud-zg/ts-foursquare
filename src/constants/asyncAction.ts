import { getType } from 'typesafe-actions'
import {
  getVenuesCategoriesAsync,
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesListedAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
} from '../DEPRECATED_actions'

interface IAsyncActionNameMapping {
  [actionType: string]: string
}

export const ASYNC_ACTION_NAME_MAPPING: IAsyncActionNameMapping = {
  // getVenuesCategoriesAsync
  [getType(getVenuesCategoriesAsync.request)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.success)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.failure)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.cancel)]: 'getVenuesCategoriesAsync',
  // getVenuesExploreAsync
  [getType(getVenuesExploreAsync.request)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.success)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.failure)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.cancel)]: 'getVenuesExploreAsync',
  // getVenuesLikesAsync
  [getType(getVenuesLikesAsync.request)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.success)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.failure)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.cancel)]: 'getVenuesLikesAsync',
  // getVenuesListedAsync
  [getType(getVenuesListedAsync.request)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.success)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.failure)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.cancel)]: 'getVenuesListedAsync',
  // getVenuesNextVenuesAsync
  [getType(getVenuesNextVenuesAsync.request)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.success)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.failure)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.cancel)]: 'getVenuesNextVenuesAsync',
  // getVenuesSearchAsync
  [getType(getVenuesSearchAsync.request)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.success)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.failure)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.cancel)]: 'getVenuesSearchAsync',
  // getVenuesSimilarAsync
  [getType(getVenuesSimilarAsync.request)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.success)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.failure)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.cancel)]: 'getVenuesSimilarAsync',
  // getVenuesSuggestCompletionAsync
  [getType(
    getVenuesSuggestCompletionAsync.request
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.success
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.failure
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.cancel
  )]: 'getVenuesSuggestCompletionAsync',
  // getVenuesTrendingAsync
  [getType(getVenuesTrendingAsync.request)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.success)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.failure)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.cancel)]: 'getVenuesTrendingAsync',
}
