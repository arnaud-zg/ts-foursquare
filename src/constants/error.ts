export enum ECommonError {
  RESPONSE_NOT_VALID = 'Response is not valid',
}
