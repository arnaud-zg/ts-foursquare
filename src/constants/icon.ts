export enum EIconSize {
  S = '32',
  M = '44',
  L = '64',
  XL = '88',
}
