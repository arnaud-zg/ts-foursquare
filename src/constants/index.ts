export const PROJECT_NAME = 'ts-foursquare'
