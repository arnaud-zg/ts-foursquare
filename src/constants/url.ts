export enum EUrlSeparator {
  LOCATION_SEARCH_SEPARATOR = '?',
  QUERY_STRING_SEPARATOR = '&',
}
