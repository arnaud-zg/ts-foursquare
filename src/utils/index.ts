export * from './icon'
export * from './url'
