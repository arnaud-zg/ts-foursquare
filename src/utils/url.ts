interface ILocationSearchParam {
  [key: string]: string | number | boolean | undefined
}

export const getLocationSearch = (
  { param = {} }: { param: ILocationSearchParam } = { param: {} }
) => {
  const locationSearch = Object.keys(param)
    .filter(
      key =>
        !(param[key] === undefined) &&
        !!(
          typeof param[key] !== 'string' ||
          (typeof param[key] === 'string' && param[key] !== '')
        )
    )
    .map(
      key => `${encodeURIComponent(key)}=${encodeURIComponent(`${param[key]}`)}`
    )
    .join('&')

  return (!!locationSearch && `?${locationSearch}`) || ''
}

export const getLocationHref = ({
  origin,
  pathname,
  param = {},
}: {
  origin: string
  pathname: string
  param?: ILocationSearchParam
}) => `${origin}${pathname}${getLocationSearch({ param })}`
