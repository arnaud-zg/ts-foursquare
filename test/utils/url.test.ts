import { EApiDefaultParameters, EApiPathnames } from '../../src/constants/api'
import { getLocationHref, getLocationSearch } from '../../src/utils/url'

describe('utils/url/getLocationSearch', () => {
  it('should return nothing', () => {
    expect(getLocationSearch()).toEqual(undefined)
    expect(getLocationSearch({ param: {} })).toEqual(undefined)
  })

  it('should generate a location search string', () => {
    expect(getLocationSearch({ param: { search: 'hello-world' } })).toEqual(
      '?search=hello-world'
    )
    expect(getLocationSearch({ param: { search: 'hello world' } })).toEqual(
      '?search=hello%20world'
    )
    expect(
      getLocationSearch({ param: { search: 'hello world', limit: 20 } })
    ).toEqual('?search=hello%20world&limit=20')
  })
})

describe('utils/url/getLocationHref', () => {
  it('should generate a location href url', () => {
    expect(
      getLocationHref({
        origin: EApiDefaultParameters.ORIGIN,
        pathname: EApiPathnames.VENUES_SEARCH,
        param: { search: 'hello-world' },
      })
    ).toMatchSnapshot()
    expect(
      getLocationHref({
        origin: EApiDefaultParameters.ORIGIN,
        pathname: EApiPathnames.VENUES_SEARCH,
        param: { search: 'hello-world', limit: 20 },
      })
    ).toMatchSnapshot()
  })
})
