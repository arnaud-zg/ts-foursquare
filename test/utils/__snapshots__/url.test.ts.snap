// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/url/getLocationHref should generate a location href url with {"city": "Séoul", "search": "한국식 바베큐"} 1`] = `"https://api.foursquare.com/v2/venues/search?city=S%C3%A9oul&search=%ED%95%9C%EA%B5%AD%EC%8B%9D%20%EB%B0%94%EB%B2%A0%ED%81%90"`;

exports[`utils/url/getLocationHref should generate a location href url with {"limit": 20, "search": "hello world"} 1`] = `"https://api.foursquare.com/v2/venues/search?search=hello%20world&limit=20"`;

exports[`utils/url/getLocationHref should generate a location href url with {"limit": 20, "search": "hello-world"} 1`] = `"https://api.foursquare.com/v2/venues/search?limit=20&search=hello-world"`;

exports[`utils/url/getLocationHref should generate a location href url with {"name": "", "search": "hello-world"} 1`] = `"https://api.foursquare.com/v2/venues/search?search=hello-world"`;

exports[`utils/url/getLocationHref should generate a location href url with {"name": undefined, "search": "hello-world"} 1`] = `"https://api.foursquare.com/v2/venues/search?search=hello-world"`;

exports[`utils/url/getLocationHref should generate a location href url with {"page": 20, "search": "hello world !"} 1`] = `"https://api.foursquare.com/v2/venues/search?page=20&search=hello%20world%20!"`;

exports[`utils/url/getLocationHref should generate a location href url with {"page": 42, "search": "hello-world"} 1`] = `"https://api.foursquare.com/v2/venues/search?page=42&search=hello-world"`;

exports[`utils/url/getLocationHref should generate a location href url with {"search": ""} 1`] = `"https://api.foursquare.com/v2/venues/search"`;

exports[`utils/url/getLocationHref should generate a location href url with {"search": "hello-world", "withOptionA": false} 1`] = `"https://api.foursquare.com/v2/venues/search?search=hello-world&withOptionA=false"`;

exports[`utils/url/getLocationHref should generate a location href url with {"search": "hello-world", "withOptionA": true} 1`] = `"https://api.foursquare.com/v2/venues/search?search=hello-world&withOptionA=true"`;

exports[`utils/url/getLocationHref should generate a location href url with {"search": "hello-world"} 1`] = `"https://api.foursquare.com/v2/venues/search?search=hello-world"`;

exports[`utils/url/getLocationHref should generate a location href url with undefined 1`] = `"https://api.foursquare.com/v2/venues/search"`;
